shell/renderer/extensions/electron_extensions_renderer_client.cc

```diff
@@ -6,16 +6,31 @@
 
 #include <string>
 
+#include "base/lazy_instance.h"
+#include "content/public/renderer/render_frame.h"
 #include "content/public/renderer/render_thread.h"
 #include "extensions/common/constants.h"
 #include "extensions/common/manifest_handlers/background_info.h"
 #include "extensions/renderer/dispatcher.h"
+#include "extensions/renderer/guest_view/mime_handler_view/mime_handler_view_container_manager.h"
 #include "shell/common/world_ids.h"
+#include "third_party/blink/public/platform/web_url.h"
+#include "third_party/blink/public/web/web_document.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "third_party/blink/public/web/web_plugin_params.h"
 
 namespace electron {
 
 ElectronExtensionsRendererClient::ElectronExtensionsRendererClient() {}
 
+// static
+ElectronExtensionsRendererClient*
```

Comment:
Are we sure this is safe ? We have access to the client instance from `renderer_client_base` via `extensions_renderer_client_`. Mixing this with lazy instance seems like a leak.

Reply:
@deepak1556 i'll check this this week - i might switch other accesses to use this same pathway to be consistent with upstream as well!